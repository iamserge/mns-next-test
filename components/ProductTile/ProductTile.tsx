import Link from 'next/link'
import Price from "../Price/Price";
import ProductImage from "../ProductImage/ProductImage";


const ProductTile = ({product}) => {
    const { 
        id,
        name,
        image_key,
        price,
        offer_ids
     } = product;
     
    return (
        <Link href={`/product/${id}`}>
            <div className="mb-10 cursor-pointer">
                <div className="relative">
                    <ProductImage image_key={image_key} name={name} offers={offer_ids} />
                    <div className="absolute"></div>
                </div>
                <div className="flex justify-between text-xl mt-2">
                    <h3>{name}</h3>
                    <Price price={price} />
                </div>
            </div>
        </Link>
    )
    
}

export default ProductTile;
