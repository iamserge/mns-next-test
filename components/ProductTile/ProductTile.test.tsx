import renderer from "react-test-renderer";
import ProductTile from "./ProductTile";

describe("ProductTile", () => {
  it("renders correctly", () => {
    const product = {
      "id": "97",
      "name": "Pure Cotton Scoop Neck Vest Top",
      "image_key": "SD_01_T41_1391_B0_X_EC_0",
      "price": {
        "current_price": 450,
        "original_price": null,
        "currency_code": "GBP"
      },
      "offer_ids": [
        "4",
        "5"
      ]
    }
    const component = renderer.create(<ProductTile product={product} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
