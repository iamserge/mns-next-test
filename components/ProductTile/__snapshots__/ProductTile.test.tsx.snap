// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTile renders correctly 1`] = `
<div
  className="mb-10 cursor-pointer"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  <div
    className="relative"
  >
    <div
      className="relative max-h-10 image-container loading-image-container"
    >
      <img
        alt="Pure Cotton Scoop Neck Vest Top"
        className="w-full min-h-full"
        loading="lazy"
        onLoad={[Function]}
        src="https://asset1.cxnmarksandspencer.com/is/image/mands/SD_01_T41_1391_B0_X_EC_0"
      />
    </div>
    <div
      className="absolute"
    />
  </div>
  <div
    className="flex justify-between text-xl mt-2"
  >
    <h3>
      Pure Cotton Scoop Neck Vest Top
    </h3>
    <div
      className="condensed"
    >
      <span
        className={null}
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "&pound;",
            }
          }
        />
        450
      </span>
    </div>
  </div>
</div>
`;
